src/pint/models/model_builder.py

```diff
@@ -24,295 +25,402 @@
+            multi_line[k[0]] += 1
+            if k[0] in self.repeatable_param:
+                repeat_par[k[0]].append(k[1:])
             else:
-                param[k[0]] = k[1:]
-        self.param_inparF = param
-        for key in repeat_par.keys():
-            self.param_inparF[key + str(1)] = self.param_inparF.pop(key)
-        pfile.close()
-        return self.param_inparF
-
-    def get_all_categories(self,):
-        """Obtain a dictionary from category to a list of instances."""
-        comp_category = defaultdict(list)
-        for k, cp in Component.component_types.items():
-            comp_category[cp.category].append(cp())
-        return dict(comp_category)
-
-    def get_comp_from_parfile(self, parfile):
-        """Right now we only have one component on each category."""
-        params_inpar = self.preprocess_parfile(parfile)
-        for cat, cmps in self.get_all_categories().items():
-            selected_c = None
-            for cpi in cmps:
-                if cpi.component_special_params:
-                    if any(par in params_inpar for par in cpi.component_special_params):
-                        selected_c = cpi
-                        # Once have match, stop searching
-                        break
-                    else:
-                        continue
+                if multi_line[k[0]] > 1:
+                    log.info("Lines with duplicate keys in par file:"
+                                 " {} and {}".format(k[0], k[1:]))
+        return param_inpar, repeat_par
+
+    def search_pulsar_system_components(self, system_name):
+        """ Search the component for the pulsar system, mostly binaries.
+        """
+        all_systems = self.category_component_map['pulsar_system']
+        result = None
+        # Search the system name first
+        print(system_name)
+        if system_name in all_systems:
+            result = self.components[system_name]
+        else: # search for the pulsar system aliases
+            for cp_name in all_systems:
+                if system_name == self.components[cp_name].binary_model_name:
+                    result = self.components[cp_name]
                 else:
-                    if cpi.is_in_parfile(params_inpar):
-                        selected_c = cpi
-            if selected_c is not None:
-                self.select_comp[cat] = selected_c
+                    continue
+        if result is None:
+            raise UnknownBinaryModel("Pulsar system/Binary model component {}"
+                                     " is not provided.".format(system_name))
+        else:
+            return result
 
-    def sort_components(self, category_order=DEFAULT_ORDER):
-        """Sort the components into order.
+    def alias_2_pint_param(self, alias):
+        """ Translate the alias to a PINT parameter name.
+        """
+        pint_par = self.param_alias_map.get(alias, None)
+        # If it is not in the map, double check if it is a repeatable par.
```

Comment:
This is confusing - isn't JUMP a repeatable par? It doesn't have prefixes. Does it use the above option or the below?

Reply:
This is for the case like "JUMP3", which is not in the aliases map. but should be recognized.